src/MultipartStreamBuilder.php

```diff
@@ -95,7 +95,14 @@ public function build()
                 $this->getHeaders($data['headers'])."\r\n";
 
             // Convert the stream to string
-            $streams .= (string) $data['contents'];
+            /* @var $contentStream StreamInterface */
+            $contentStream = $data['contents'];
+            if ($contentStream->isSeekable()) {
+                $streams .= (string) $data['contents'];
```

Comment:
Great to hear the PR will be merged next week. I agree with your comment about readability and have made the change from string casting to explicitly calling __toString().